src/import/sst_service.rs

```diff
@@ -93,23 +93,48 @@ const WRITER_GC_INTERVAL: Duration = Duration::from_secs(300);
 const SUSPEND_REQUEST_MAX_SECS: u64 = // 6h
     6 * 60 * 60;
 
+const REJECT_SERVE_MEMORY_USAGE: u64 = 1024 * 1024 * 1024; //1G
+// consider block cache and raft store. the memory usage will be
+const HIGH_IMPORT_MEMORY_WATER_RATIO: f64 = 0.95;
+
 /// Check if the system has enough resources for import tasks
 fn check_import_resources() -> Result<()> {
     // Check disk space first
     if get_disk_status(0) != DiskUsage::Normal {
         return Err(Error::DiskSpaceNotEnough);
     }
 
-    // Check memory usage
-    // high water is 90% of the memory limit by default
-    let mut usage = 0;
-    if memory_usage_reaches_high_water(&mut usage) {
+    let usage = get_global_memory_usage();
+    let mem_limit = (|| {
+        fail_point!("mock_memory_limit", |t| {
+            t.unwrap().parse::<u64>().unwrap()
+        });
+        SysQuota::memory_limit_in_bytes()
+    })();
+
+    if mem_limit == 0 || mem_limit < usage {
```

Comment:
it seems mem_limit could be -1 if unlimited when using cgroup. maybe can have better logging for that case?

Reply:
mem_limit is u64, so I think it cannot be -1